Data-Structure/Algorithm-Design-Part-1/DictionaryUsingBST/TheKeyValuePairIsNullException.java

```diff
@@ -0,0 +1,17 @@
+package DictionaryUsingBST;
+
+/**
+ * @author Niraj Gupta
+ * Class to represent the exception when the either/both the key or value pair are/is null.
+ *
+ */
+public class TheKeyValuePairIsNullException extends NullPointerException{

```

Comment:
We do not need a custom exception to handle null pointer case explicitly. If in any case the null case is left unhandled system autothrows null pointer exception. Our job is to design a robust system so that we never get such exception.